tests/vision/detection/test_data.py

```diff
@@ -0,0 +1,126 @@
+import json
+import os
+from pathlib import Path
+
+import pytest
+import torch
+from PIL import Image
+from torchvision import transforms as T
+
+from flash.vision.detection.data import ImageDetectionData
+
+try:
+    from pycocotools.coco import COCO
+except ImportError:
+    COCO = None
+
+
+def _create_dummy_coco_json(dummy_json_path):
+
+    dummy_json = {
+        "images": [{
+            "id": 0,
+            'width': 500,
+            'height': 500,
+            'file_name': 'sample_one.png',
+        }, {
+            "id": 1,
+            "width": 500,
+            "height": 500,
+            "file_name": "sample_two.png",
+        }],
+        "annotations": [{
+            "id": 1,
+            "image_id": 0,
+            "category_id": 0,
+            "area": 150,
+            "bbox": [30, 40, 20, 20],
+            "iscrowd": 0,
+        }, {
+            "id": 2,
+            "image_id": 1,
+            "category_id": 0,
+            "area": 240,
+            "bbox": [50, 100, 280, 15],
+            "iscrowd": 0,
+        }, {
+            "id": 3,
+            "image_id": 1,
+            "category_id": 0,
+            "area": 170,
+            "bbox": [230, 130, 90, 180],
+            "iscrowd": 0,
+        }],
+        "categories": [{
+            "id": 0,
+            "name": "person",
+            "supercategory": "person",
+        }]
+    }
+
+    with open(dummy_json_path, "w") as fp:
+        json.dump(dummy_json, fp)
+
+
+def _create_synth_coco_dataset(tmpdir):
+    train_dir = Path(tmpdir / "train")
+    train_dir.mkdir()
+
+    (train_dir / "images").mkdir()
+    Image.new('RGB', (500, 500)).save(train_dir / "images" / "sample_one.png")
```

Comment:
Do you need the images to be 500 x 500 ?

Reply:
@tchaton Nope. Just dummy numbers for testing

Reply:
Can you test with non-squared images then? Size doesn't matter here. Maybe I'd use the Full-HD resolution since this one is quite common

Reply:
Makes sense, will update it to `1920 x 1080`